language-server/server.go

```diff
@@ -138,6 +186,120 @@ func NewServer(version string, lintRequest *utils.LintFileRequest) *ServerState
 	return state
 }
 
+func (s *ServerState) applyAutoFix(
+	doc *Document,
+	ruleId string,
+	diagnosticRange protocol.Range,
+) (string, protocol.Range) {
+	rule, exists := s.lintRequest.SelectedRS.Rules[ruleId]
+	if !exists || rule.AutoFixFunction == "" {
+		return "", diagnosticRange
+	}
+
+	autoFixFunc, exists := s.lintRequest.AutoFixFunctions[rule.AutoFixFunction]
+	if !exists {
+		return "", diagnosticRange
+	}
+
+	var rootNode yaml.Node
+	if err := yaml.Unmarshal([]byte(doc.Content), &rootNode); err != nil {
+		return "", diagnosticRange
+	}
+
+	// Find the node at the diagnostic location
+	targetLine := int(diagnosticRange.Start.Line) + 1
```

Comment:
What about value nodes that are on the same line as the key, like `thing: thang`